lighthouse/src/main.rs

```diff
@@ -126,37 +127,39 @@ fn main() {
         .subcommand(account_manager::cli_app())
         .get_matches();
 
-    // boot node subcommand circumvents the environment
-    if let Some(bootnode_matches) = matches.subcommand_matches("boot_node") {
-        // The bootnode uses the main debug-level flag
-        let debug_info = matches
-            .value_of("debug-level")
-            .expect("Debug-level must be present")
-            .into();
-        boot_node::run(bootnode_matches, debug_info);
-        return;
-    }
-
     // Debugging output for libp2p and external crates.
     if matches.is_present("env_log") {
         Builder::from_env(Env::default()).init();
     }
 
-    macro_rules! run_with_spec {
-        ($env_builder: expr) => {
-            run($env_builder, &matches)
-        };
-    }
+    let result = load_testnet_config(&matches).and_then(|testnet_config| {
+        let eth_spec_id = testnet_config.eth_spec_id()?;
+
+        // boot node subcommand circumvents the environment
+        if let Some(bootnode_matches) = matches.subcommand_matches("boot_node") {
+            // The bootnode uses the main debug-level flag
+            let debug_info = matches
+                .value_of("debug-level")
+                .expect("Debug-level must be present")
+                .into();
 
-    let result = match matches.value_of("spec") {
-        Some("minimal") => run_with_spec!(EnvironmentBuilder::minimal()),
-        Some("mainnet") => run_with_spec!(EnvironmentBuilder::mainnet()),
-        Some("interop") => run_with_spec!(EnvironmentBuilder::interop()),
-        spec => {
-            // This path should be unreachable due to slog having a `default_value`
-            unreachable!("Unknown spec configuration: {:?}", spec);
+            boot_node::run(bootnode_matches, eth_spec_id, debug_info);
+
+            return Ok(());
         }
-    };
+
+        macro_rules! run_with_spec {
+            ($env_builder: expr) => {
+                run($env_builder, &matches, testnet_config)
+            };
+        }
```

Comment:
I don't think this doesn't need to be a macro, but I don't mind if it stays :man_shrugging: (I thought I'd deleted it :P )